packages/fuels/src/setup-launch-node-server.ts

```diff
@@ -0,0 +1,142 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
+/* eslint-disable no-console */
+import type { LaunchNodeOptions } from '@fuel-ts/account/test-utils';
+import { launchNode } from '@fuel-ts/account/test-utils';
+import { waitUntilUnreachable } from '@fuel-ts/utils/test-utils';
+import http from 'http';
+import type { AddressInfo } from 'net';
+
+process.setMaxListeners(Infinity);
+
+async function parseBody(req: http.IncomingMessage): Promise<LaunchNodeOptions> {
+  return new Promise<LaunchNodeOptions>((resolve, reject) => {
+    const body: Buffer[] = [];
+    req.on('data', (chunk) => {
+      body.push(chunk);
+    });
+    req.on('end', () => {
+      try {
+        resolve(JSON.parse(body.length === 0 ? '{}' : Buffer.concat(body).toString()));
+      } catch (err) {
+        reject(err);
+      }
+    });
+    req.on('error', reject);
+  });
+}
+
+const cleanupFns: Map<string, () => Promise<void>> = new Map();
+
+const server = http.createServer(async (req, res) => {
+  res.setHeader('Access-Control-Allow-Origin', '*');
+
+  if (req.url === '/') {
+    try {
+      const body = await parseBody(req);
+
+      const node = await launchNode({
+        port: '0',
+        loggingEnabled: false,
+        debugEnabled: false,
+        ...body,
+        fuelCorePath: 'fuels-core',
+      });
+
+      cleanupFns.set(node.url, async () => {
+        node.cleanup();
+        await waitUntilUnreachable(node.url);
+        cleanupFns.delete(node.url);
+      });
+
+      res.end(node.url);
+    } catch (err) {
+      console.error(err);
+      res.writeHead(500, { 'Content-Type': 'text/plain' });
+      res.end(JSON.stringify(err));
+    }
+    return;
+  }
+
+  if (req.url?.startsWith('/cleanup')) {
+    const nodeUrl = req.url?.match(/\/cleanup\/(.+)/)?.[1];
+    if (nodeUrl) {
+      const cleanupFn = cleanupFns.get(nodeUrl);
+      await cleanupFn?.();
+      res.end();
+    }
+  }
+});
+
+function closeServer() {
+  return new Promise<void>((resolve) => {
```

Comment:
Nit I.

(written via the UI, requires double-check)

```suggestion
const closeServer = (event: string) => (error?: unknown) => {
  console.log(event);
  if (error) {
    console.log(error);
  }
  return new Promise<void>((resolve) => {
```

Reply:
https://github.com/FuelLabs/fuels-ts/pull/2378/commits/9d281fd8a8f3d3e54005b9b6498d51f7749c22a3